esp/esdllib/esdl_script.cpp

```diff
@@ -78,27 +116,99 @@ class CEsdlScriptContext : public CInterfaceOf<IEsdlScriptContext>
             tracerRef().uwarnlog("enable masking request not completed - no masking engine");
             return false;
         }
-        if (maskingEnabled())
+        if (!maskerScopes.empty() && maskerScopes.back())
         {
             domainId = (isEmptyString(domainId) ? maskingEngine->inspector().queryDefaultDomain() : domainId);
             version = (0 == version ? maskingEngine->inspector().queryDefaultVersion() : version);
-            IDataMaskingProfileContext* ctx = maskerStack.front();
-            if (ctx->inspector().acceptsDomain(domainId) && ctx->queryVersion() == version)
+            IDataMaskingProfileContext* masker = maskerScopes.back();
+            if (masker->inspector().acceptsDomain(domainId) && masker->queryVersion() == version)
                 return true;
-            tracerRef().uwarnlog("enable masking request using %s:%hhu not completed - already enabled using %s:%hhu", domainId, version, ctx->queryDomain(), ctx->queryVersion());
+            tracerRef().uwarnlog("enable masking request using %s:%hhu not completed - already enabled using %s:%hhu", domainId, version, masker->queryDomain(), masker->queryVersion());
             return false;
         }
-        Owned<IDataMaskingProfileContext> ctx(maskingEngine->getContext(domainId, version, tracer));
-        if (!ctx)
+        Owned<IDataMaskingProfileContext> masker(maskingEngine->getContext(domainId, version, tracer));
+        if (!masker)
         {
             tracerRef().uwarnlog("enable masking request not completed - no context for '%s:%hhu'", domainId, version);
             return false;
         }
-        maskerStack.emplace_back(ctx.getClear());
-        return maskingEnabled();
+        if (maskerScopes.empty())
+            maskerScopes.emplace_back(masker.getLink());
```

Comment:
Curious why you don't consistently use getClear() in both places?  both getClear() and getLink() return ownership.

Reply:
Since both are functionally correct, added a comment to https://track.hpccsystems.com/browse/HPCC-29156 to include this in the general cleanup.